data-prepper-api/src/main/java/org/opensearch/dataprepper/model/source/coordinator/SourcePartition.java

```diff
@@ -15,18 +15,18 @@
  */
 public class SourcePartition<T> {
```

Comment:
There may need to be more metadata in this object. Info about who owns it and when they reserved it, etc. Even for a future non-lease based approach, knowing which node tried to process it and when would be helpful for debugging.

Reply:
This object is supplied to the sources. I could see value in providing reserve time and maybe the time that this partition ownership will timeout.

However, by definition if this is returned to the source then the owner is this instance, and I don't think the source should concern itself with this. the core source coordination logic will always be tracking and logging where needed who owns a partition and who processes it